src/display_servers/xlib_display_server/event_translate.rs

```diff
@@ -103,6 +106,20 @@ fn from_map_request(raw_event: xlib::XEvent, xw: &XWrap) -> Option<DisplayEvent>
     Some(DisplayEvent::WindowCreate(w, cursor.0, cursor.1))
 }
 
+fn from_mapping_notify(raw_event: xlib::XEvent, xw: &XWrap) -> Option<DisplayEvent> {
+    let mut event = xlib::XMappingEvent::from(raw_event);
+    if event.request == xlib::MappingModifier || event.request == xlib::MappingKeyboard {
+        // refresh keyboard
+        log::info!("Updating keyboard");
+        xw.refresh_keyboard(&mut event).unwrap();
+
```

Comment:
Instead of unwrap you should be able to use ?, e.g ```xw.refresh_keyboard(&mut event)?```.
As unwrap will lead to panics if there is an error.